src/mapblock.cpp

```diff
@@ -549,6 +613,18 @@ void MapBlock::deSerialize(std::istream &is, u8 version, bool disk)
 			m_node_timers.deSerialize(is, version);
 		}
 	}
+	if (version >= 29) {
+		try {
+			TRACESTREAM(<<"MapBlock::deSerialize "<<PP(getPos())
+				<<": Node metadata"<<std::endl);
+
+			m_node_metadata.deSerialize(is, m_gamedef->idef());
+		} catch(SerializationError &e) {
+			warningstream<<"MapBlock::deSerialize(): Ignoring an error"
+				<<" while deserializing node metadata at ("
+				<<PP(getPos())<<": "<<e.what()<<std::endl;
+		}
+	}
```

Comment:
Hm, so metadata is now at the end?
Not that it really matters but I expected it to remain where it was before.

Reply:
Oh... I thought that's what we discussed. It's at the end so that you can ignore it. Maybe I misunderstood. I don't care whether it's at the end, as long as it works. :)

Reply:
Well with the node-id mapping now moved before the node data you can simply stop reading there, no need for the meta to be moved.
If you don't mind I could checkout your branch and push a commit moving it back.

Reply:
I'll do it. NP.